controller/namedusers.go

```diff
@@ -0,0 +1,64 @@
+package controller
+
+import (
+	"github.com/fabric8-services/fabric8-auth/account/service"
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/application"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/jsonapi"
+	"github.com/fabric8-services/fabric8-auth/log"
+	"github.com/fabric8-services/fabric8-auth/token"
+
+	"github.com/goadesign/goa"
+)
+
+// NamedusersController implements the namedusers resource.
+type NamedusersController struct {
+	*goa.Controller
+	app           application.Application
+	config        UsersControllerConfiguration
+	tenantService service.Tenant
+}
+
+// NewNamedusersController creates a namedusers controller.
+func NewNamedusersController(service *goa.Service, app application.Application, config UsersControllerConfiguration, tenantService service.Tenant) *NamedusersController {
+	return &NamedusersController{
+		Controller:    service.NewController("NamedusersController"),
+		app:           app,
+		config:        config,
+		tenantService: tenantService,
+	}
+}
+
+// Deprovision runs the deprovision action.
+func (c *NamedusersController) Deprovision(ctx *app.DeprovisionNamedusersContext) error {
+	isSvcAccount := token.IsSpecificServiceAccount(ctx, token.OnlineRegistration)
+	if !isSvcAccount {
+		log.Error(ctx, nil, "the account is not an authorized service account allowed to deprovision users")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("account not authorized to deprovision users"))
+	}
+
+	identity, err := c.app.UserService().DeprovisionUser(ctx, ctx.Username)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":      err,
+			"username": ctx.Username,
+		}, "unable to deprovision user")
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	// Delete tenant
+	if c.tenantService != nil {
+		err := c.tenantService.Delete(ctx, identity.ID)
+		if err != nil {
+			log.Error(ctx, map[string]interface{}{
+				"err":         err,
+				"identity_id": identity.ID,
+				"username":    ctx.Username,
+			}, "unable to delete tenant when deprovisioning user")
+			// Just log the error and proceed
```

Comment:
This looks right. Is there something we could do to make this show up on sentry ? Would be nice to see and track the errors here.

Reply:
Good idea. Added sentry logging in https://github.com/fabric8-services/fabric8-auth/pull/524/commits/478509453d11e35b93d1d7ae5e8c0d06e5454d13